apps/projection/src/app/ui/card/card.component.ts

```diff
@@ -1,21 +1,46 @@
-import { NgFor, NgIf } from '@angular/common';
-import { Component, Input } from '@angular/core';
+import { NgClass, NgFor, NgIf, NgTemplateOutlet } from '@angular/common';
+import {
+  Component,
+  ContentChild,
+  Directive,
+  HostBinding,
+  Input,
+  TemplateRef,
+} from '@angular/core';
 import { randStudent, randTeacher } from '../../data-access/fake-http.service';
 import { StudentStore } from '../../data-access/student.store';
 import { TeacherStore } from '../../data-access/teacher.store';
 import { CardType } from '../../model/card.model';
-import { ListItemComponent } from '../list-item/list-item.component';
+
+@Directive({ selector: 'ng-template[card-list]', standalone: true })
+export class CardListDirective {}
 
 @Component({
   selector: 'app-card',
   templateUrl: './card.component.html',
   standalone: true,
-  imports: [NgIf, NgFor, ListItemComponent],
+  imports: [NgIf, NgFor, NgTemplateOutlet, NgClass],
+  styles: [
+    `
+      .bg-light-green {
+        background-color: rgba(0, 250, 0, 0.1);
+      }
+
+      .bg-light-red {
+        background-color: rgba(250, 0, 0, 0.1);
+      }
+    `,
+  ],
 })
 export class CardComponent {
+  @Input() isTeacher = false;
+  @Input() isStudent = false;
+
+  @ContentChild(CardListDirective, { read: TemplateRef })
+  cardList!: TemplateRef<any>;
```

Comment:
you can type the Template instead of using any.

Reply:
please can you elaborate more on this for me?

Reply:
You can add an interface instead of any to tell what context the user need to set